README.md

```diff
@@ -1,73 +1,61 @@
-<a href="https://circleci.com/gh/nestjs/nest" target="_blank"><img src="https://img.shields.io/circleci/build/github/nestjs/nest/master" alt="CircleCI" /></a>
-<a href="https://coveralls.io/github/nestjs/nest?branch=master" target="_blank"><img src="https://coveralls.io/repos/github/nestjs/nest/badge.svg?branch=master#9" alt="Coverage" /></a>
-<a href="https://discord.gg/G7Qnnhy" target="_blank"><img src="https://img.shields.io/badge/discord-online-brightgreen.svg" alt="Discord"/></a>
-<a href="https://opencollective.com/nest#backer" target="_blank"><img src="https://opencollective.com/nest/backers/badge.svg" alt="Backers on Open Collective" /></a>
-<a href="https://opencollective.com/nest#sponsor" target="_blank"><img src="https://opencollective.com/nest/sponsors/badge.svg" alt="Sponsors on Open Collective" /></a>
-  <a href="https://paypal.me/kamilmysliwiec" target="_blank"><img src="https://img.shields.io/badge/Donate-PayPal-ff3f59.svg"/></a>
-    <a href="https://opencollective.com/nest#sponsor"  target="_blank"><img src="https://img.shields.io/badge/Support%20us-Open%20Collective-41B883.svg" alt="Support us"></a>
-  <a href="https://twitter.com/nestframework" target="_blank"><img src="https://img.shields.io/twitter/follow/nestframework.svg?style=social&label=Follow"></a>
-</p>
-  <!--[![Backers on Open Collective](https://opencollective.com/nest/backers/badge.svg)](https://opencollective.com/nest#backer)
-  [![Sponsors on Open Collective](https://opencollective.com/nest/sponsors/badge.svg)](https://opencollective.com/nest#sponsor)-->
-
-## Description
-
-[Nest](https://github.com/nestjs/nest) framework TypeScript starter repository.
-
-## Installation
-
-```bash
-$ npm install
+# Nestjs Transporter For NSQ
+
+## Overview
+
+This library provides a nestjs transporter by taking NSQ as its message broker. We could simply take the concept of a transporter as an event-driven framework as follows:
+
+- As a message producer, I could send a message to NSQ to various topics by emitting various events based on the topic names.
+- As a message consumer, I could receive a message by subscribing to those emitted events.
+
+For more details on the overal
+architecture, please kinldy refer to this [article](https://dev.to/nestjs/integrate-nestjs-with-external-services-using-microservice-transporters-part-1-p3)
+
+## How to use it
+
+### Create a dynamic module called NsqModule as follows:
+
+```typescript
+import { DynamicModule, Module } from '@nestjs/common';
+import { ClientProxy } from '@nestjs/microservices';
+import { ClientNsq, NsqOptions } from 'nestjs-nsq-transporter';
+
+@Module({})
+export class NsqModule {
+  static configure(options: { client: NsqOptions }): DynamicModule {
+    return {
+      module: NsqModule,
+      providers: [
+        {
+          provide: 'NSQ_CLIENT',
+          useFactory: (): ClientProxy => new ClientNsq(options.client),
```

Comment:
the `options` is not injected in factory function.
There are many fields in the `NsqOptions`, e.g. `requeueDelay `, `maxAttempts `, some have default value, and the time unit is not clear (second or millisecond). 
I suggest document all of them in readme so that developers won't need to check the code to find the options.

Reply:
Updated in the latest commit.